### PR TITLE
Add support for transferring inter-admin-unit tasks.

### PR DESCRIPTION
**Pull request by lukasgraf**
This adds support for also transferring inter-admin-unit tasks using the `@transfer-tasks` endpoint.

For changing the `responsible`, the existing "reassign task" mechanism, which involves an actual workflow transition, is reused. For changing the `issuer`, that mechanism is emulated, and supported by a new `IssuerChangeResponseSyncerSender` that syncs the change to the remote side (without triggering an actual WF transition though, because there is none).

The situation for forwardings doesn't really change with the support for inter-admin-unit tasks, because forwardings can never be in a state where the predecessor forwarding is still open - therefore, no syncing of changes is required. 

## Checklist (Must have)

- [x] Changelog entry
- [x] Link to issue (Jira or GitHub) and backlink in issue (Jira)

Jira: https://4teamwork.atlassian.net/browse/GEVER-541

## Checklist (if applicable)

- API change:
  - [x] Documentation is updated
  _(Not needed, checked)_

- New functionality:
  - [x] for `task` also works for `forwarding`

### CODE REVIEW

File: opengever/api/transfer.py [hunk 1]
```diff
@@ -62,41 +72,100 @@ def reply(self):
...
+
+        self.context.issuer = new_issuer
+        notify(ObjectModifiedEvent(self.context))
+        TaskChangeIssuerActivity(self.context, self.context.REQUEST, issuer_response).record()
+
+        sync_task_response(
+            self.context, self.request, 'issuer-change',
+            transition, text=None, new_issuer=new_issuer)
+
+    def create_watcher_and_commit_if_needed(self, new_responsible):
```
Comment (deiferni) [inline opengever/api/transfer.py:135]: This is an exact a copy of what we already have in `AssignTaskForm`? If so i would prefer to have this critical path extracted into e.g. a method and reused by both the `AssignTaskForm` and `TransferTaskPost`. Maybe a function in `opengever.activity` would be a good place to put it?

File: opengever/task/response_syncer/issuer_change.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    def raise_sync_exception(self, task, transition, text, **kwargs):
+        raise ResponseSyncerSenderException(
+            'Could not transfer issuer for task on remote '
+            'admin unit {} ({})'.format(
+                task.admin_unit_id,
+                task.physical_path))
+
+    def _is_synced_transition(self, transition):
+        return transition in [
+            'task-transition-change-issuer',
```
Comment (deiferni) [inline opengever/task/response_syncer/issuer_change.py:37]: Would it make sense to extract this state into one of the constants in https://github.com/4teamwork/opengever.core/blob/abed7d9520e21478f312ab18079f40cd5d0fbe3c/opengever/task/__init__.py ?

File: opengever/testing/test_case.py [hunk 3]
```diff
@@ -76,6 +78,44 @@ def allowed(obj):
         children['added'] = set(children['after']) - set(children['before'])
         children['removed'] = set(children['before']) - set(children['after'])
 
+    @contextmanager
+    def observe_responses(self, obj):
```
Comment (deiferni) [inline opengever/testing/test_case.py:82]: 💛

**Review by deiferni — APPROVED** [review]
LGTM 👍 , just two remarks. I also like the new `observe_[thingy]` context managers 🎉 .

File: opengever/api/transfer.py (hunk 1 above)
Comment (lukasgraf) [reply opengever/api/transfer.py:135]: I deliberately did not factor that out, because that would move the code far away from the place where it's being used, and make it much less obvious that something rather unusual is being done in these two places. Having the `transaction.commit(); transaction.begin()` (which should raise a couple flags when spotted) right there in the same file, next to the context where it's being used, trumps DRY / code reuse by far here IMHO.

In other words: I don't trust people enough to responsibly use that function, that's why I don't want to factor it out. It _should_ not be reusable with minimal effort. If we actually do require this pattern in a third place, then we should consider it, but for now it's strictly tied to the "reassign" mechanics (and documented as such).

File: opengever/task/response_syncer/issuer_change.py (hunk 2 above)
Comment (lukasgraf) [reply opengever/task/response_syncer/issuer_change.py:37]: I don't think so. It's a pseudo-transition, as mentioned in the class below for the receiving side. And it has specifically been invented to facilitate that issuer-change via the `@transfer-task` endpoint. I see zero chance of this being reused anytime soon.
